cockpit-container-update

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+# Update .cockpit-ci/container to the latest tasks container tag automatically
+
+import json
+import os
+import socket
+import sys
+import urllib.parse
+import urllib.request
+from ssl import SSLEOFError
+
+import task
+from lib.constants import BASE_DIR
+
+sys.dont_write_bytecode = True
+
+
+URL = 'https://quay.io/api/v1/repository/cockpit/tasks/tag/?onlyActiveTags=true'
+
+
+def fetch_tags(url):
+    try:
+        req = urllib.request.Request(url)
+        with urllib.request.urlopen(req) as response:
+            return json.loads(response.read().decode().strip())
+    except (urllib.error.HTPPError, ConnectionResetError, socket.error, SSLEOFError) as e:
+        print("Transient error during quay.io request %s", e)
+
+    return None
+
+
+def run(context, verbose=False, **kwargs):
+    container = context or os.path.join('.cockpit-ci', 'container')
+    container_path = os.path.join(BASE_DIR, container)
+
+    with open(container_path) as fp:
+        content = fp.read()
+
+    current_tag = content.strip().partition(':')[-1]
+    tags = fetch_tags(URL)
+    if tags is None:
+        return
+
+    # Strip latest tags
+    tags = [t for t in tags['tags'] if t['name'] != 'latest']
+    latest_image = sorted(tags, key=lambda x: x['name'])[-1]
+    latest_tag = latest_image['name']
+
+    new_content = content.replace(current_tag, latest_tag)
```

Comment:
This is buggy if the tags could be substrings of one another. Let's take the container image name from the original `.partition()` call and append the new tag.